NPE occurs  when  execute show table status

## Bug Report

### Which version of ShardingSphere did you use?
5.5.1-SNAPSHOT master e55969073952e82c9192f1876d48bec3dba511c6

### Which project did you use? ShardingSphere-JDBC or ShardingSphere-Proxy?
ShardingSphere-Proxy

### Actual behavior
- Client
```plain
mysql> show table status;
ERROR 30000 (HY000): Unknown exception.
More details: java.lang.NullPointerException: Cannot invoke "java.math.BigInteger.add(java.math.BigInteger)" because "num1" is null
mysql> 
```

- Proxy log
```plain
[INFO ] 2024-08-26 18:04:06.880 [ShardingSphere-Command-0] ShardingSphere-SQL - Logic SQL: show table status
[INFO ] 2024-08-26 18:04:06.880 [ShardingSphere-Command-0] ShardingSphere-SQL - Actual SQL: ds_1 ::: show table status
[INFO ] 2024-08-26 18:04:06.880 [ShardingSphere-Command-0] ShardingSphere-SQL - Actual SQL: ds_0 ::: show table status
[ERROR] 2024-08-26 18:04:07.089 [ShardingSphere-Command-0] o.a.s.p.f.c.CommandExecutorTask - Exception occur: 
java.lang.NullPointerException: Cannot invoke "java.math.BigInteger.add(java.math.BigInteger)" because "num1" is null
	at org.apache.shardingsphere.sharding.merge.dal.show.ShowTableStatusMergedResult.sum(ShowTableStatusMergedResult.java:77)
	at org.apache.shardingsphere.sharding.merge.dal.show.ShowTableStatusMergedResult.merge(ShowTableStatusMergedResult.java:68)
	at org.apache.shardingsphere.sharding.merge.dal.show.ShowTableStatusMergedResult.init(ShowTableStatusMergedResult.java:58)
	at org.apache.shardingsphere.sharding.merge.dal.show.ShowTableStatusMergedResult.init(ShowTableStatusMergedResult.java:39)
	at org.apache.shardingsphere.infra.merge.result.impl.memory.MemoryMergedResult.<init>(MemoryMergedResult.java:57)
	at org.apache.shardingsphere.sharding.merge.dal.show.ShowTableStatusMergedResult.<init>(ShowTableStatusMergedResult.java:43)
	at org.apache.shardingsphere.sharding.merge.dal.ShardingDALResultMerger.merge(ShardingDALResultMerger.java:71)
	at org.apache.shardingsphere.infra.merge.MergeEngine.executeMerge(MergeEngine.java:88)
	at org.apache.shardingsphere.infra.merge.MergeEngine.merge(MergeEngine.java:78)
	at org.apache.shardingsphere.proxy.backend.connector.DatabaseConnector.mergeQuery(DatabaseConnector.java:320)
	at org.apache.shardingsphere.proxy.backend.connector.DatabaseConnector.processExecuteQuery(DatabaseConnector.java:290)
	at org.apache.shardingsphere.proxy.backend.connector.DatabaseConnector.doExecute(DatabaseConnector.java:251)
	at org.apache.shardingsphere.proxy.backend.connector.DatabaseConnector.execute(DatabaseConnector.java:199)
	at org.apache.shardingsphere.proxy.frontend.mysql.command.query.text.query.MySQLComQueryPacketExecutor.execute(MySQLComQueryPacketExecutor.java:86)
	at org.apache.shardingsphere.proxy.frontend.command.CommandExecutorTask.doExecuteCommand(CommandExecutorTask.java:126)
	at org.apache.shardingsphere.proxy.frontend.command.CommandExecutorTask.executeCommand(CommandExecutorTask.java:121)
	at org.apache.shardingsphere.proxy.frontend.command.CommandExecutorTask.run(CommandExecutorTask.java:78)
```

--- a/features/sharding/core/src/main/java/org/apache/shardingsphere/sharding/merge/dal/show/ShowTableStatusMergedResult.java
+++ b/features/sharding/core/src/main/java/org/apache/shardingsphere/sharding/merge/dal/show/ShowTableStatusMergedResult.java
@@ -88,7 +88,7 @@ public final class ShowTableStatusMergedResult extends MemoryMergedResult<Shardi
         return BigInteger.ZERO.equals(numberBigInteger) ? BigInteger.ZERO : getNonNullBigInteger(sum).divide(numberBigInteger);
     }
 
-    private static BigInteger getNonNullBigInteger(final Object value) {
+    private BigInteger getNonNullBigInteger(final Object value) {
         return Optional.ofNullable(value)
                 .filter(BigInteger.class::isInstance)
                 .map(BigInteger.class::cast)

--- a/features/sharding/core/src/main/java/org/apache/shardingsphere/sharding/merge/dal/show/ShowTableStatusMergedResult.java
+++ b/features/sharding/core/src/main/java/org/apache/shardingsphere/sharding/merge/dal/show/ShowTableStatusMergedResult.java
@@ -87,7 +87,7 @@ public final class ShowTableStatusMergedResult extends MemoryMergedResult<Shardi
         BigInteger numberBigInteger = getNonNullBigInteger(number);
         return BigInteger.ZERO.equals(numberBigInteger) ? BigInteger.ZERO : getNonNullBigInteger(sum).divide(numberBigInteger);
     }
-
+    
     private BigInteger getNonNullBigInteger(final Object value) {
         return Optional.ofNullable(value)
                 .filter(BigInteger.class::isInstance)

--- a/features/sharding/core/src/main/java/org/apache/shardingsphere/sharding/merge/dal/show/ShowTableStatusMergedResult.java
+++ b/features/sharding/core/src/main/java/org/apache/shardingsphere/sharding/merge/dal/show/ShowTableStatusMergedResult.java
@@ -74,10 +74,16 @@ public final class ShowTableStatusMergedResult extends MemoryMergedResult<Shardi
     }
     
     private BigInteger sum(final Object num1, final Object num2) {
+        if (num1 == null || num2 == null){
+            return BigInteger.ZERO;
+        }
         return ((BigInteger) num1).add((BigInteger) num2);
     }
     
     private BigInteger avg(final Object sum, final Object number) {
+        if (sum == null || number == null){
+            return BigInteger.ZERO;
+        }
         return BigInteger.ZERO.equals(number) ? BigInteger.ZERO : ((BigInteger) sum).divide((BigInteger) number);
     }
 }

--- a/features/sharding/core/src/main/java/org/apache/shardingsphere/sharding/merge/dal/show/ShowTableStatusMergedResult.java
+++ b/features/sharding/core/src/main/java/org/apache/shardingsphere/sharding/merge/dal/show/ShowTableStatusMergedResult.java
@@ -17,6 +17,7 @@
 
 package org.apache.shardingsphere.sharding.merge.dal.show;
 
+import com.google.common.base.MoreObjects;
 import org.apache.shardingsphere.infra.binder.context.statement.SQLStatementContext;
 import org.apache.shardingsphere.infra.executor.sql.execute.result.query.QueryResult;
 import org.apache.shardingsphere.infra.merge.result.impl.memory.MemoryMergedResult;
@@ -74,16 +75,24 @@ public final class ShowTableStatusMergedResult extends MemoryMergedResult<Shardi
     }
     
     private BigInteger sum(final Object num1, final Object num2) {
-        if (num1 == null || num2 == null) {
+        if (num1 == null && num2 == null) {
             return null;
         }
-        return ((BigInteger) num1).add((BigInteger) num2);
+        return getNonNullBigInteger(num1).add(getNonNullBigInteger(num2));
     }
     
     private BigInteger avg(final Object sum, final Object number) {
-        if (sum == null || number == null) {
+        if (sum == null && number == null) {
             return null;
         }
-        return BigInteger.ZERO.equals(number) ? BigInteger.ZERO : ((BigInteger) sum).divide((BigInteger) number);
+        BigInteger numberBigInteger = getNonNullBigInteger(number);
+        return BigInteger.ZERO.equals(numberBigInteger) ? BigInteger.ZERO : getNonNullBigInteger(sum).divide(numberBigInteger);
+    }
+
+    private static BigInteger getNonNullBigInteger(Object value) {
+        return Optional.ofNullable(value)
+                .filter(BigInteger.class::isInstance)
+                .map(BigInteger.class::cast)
+                .orElse(BigInteger.ZERO);
     }
 }

--- a/features/sharding/core/src/main/java/org/apache/shardingsphere/sharding/merge/dal/show/ShowTableStatusMergedResult.java
+++ b/features/sharding/core/src/main/java/org/apache/shardingsphere/sharding/merge/dal/show/ShowTableStatusMergedResult.java
@@ -75,14 +75,14 @@ public final class ShowTableStatusMergedResult extends MemoryMergedResult<Shardi
     
     private BigInteger sum(final Object num1, final Object num2) {
         if (num1 == null || num2 == null){
-            return BigInteger.ZERO;
+            return null;
         }
         return ((BigInteger) num1).add((BigInteger) num2);
     }
     
     private BigInteger avg(final Object sum, final Object number) {
         if (sum == null || number == null){
-            return BigInteger.ZERO;
+            return null;
         }
         return BigInteger.ZERO.equals(number) ? BigInteger.ZERO : ((BigInteger) sum).divide((BigInteger) number);
     }

--- a/features/sharding/core/src/main/java/org/apache/shardingsphere/sharding/merge/dal/show/ShowTableStatusMergedResult.java
+++ b/features/sharding/core/src/main/java/org/apache/shardingsphere/sharding/merge/dal/show/ShowTableStatusMergedResult.java
@@ -74,14 +74,14 @@ public final class ShowTableStatusMergedResult extends MemoryMergedResult<Shardi
     }
     
     private BigInteger sum(final Object num1, final Object num2) {
-        if (num1 == null || num2 == null){
+        if (num1 == null || num2 == null) {
             return null;
         }
         return ((BigInteger) num1).add((BigInteger) num2);
     }
     
     private BigInteger avg(final Object sum, final Object number) {
-        if (sum == null || number == null){
+        if (sum == null || number == null) {
             return null;
         }
         return BigInteger.ZERO.equals(number) ? BigInteger.ZERO : ((BigInteger) sum).divide((BigInteger) number);

--- a/features/sharding/core/src/main/java/org/apache/shardingsphere/sharding/merge/dal/show/ShowTableStatusMergedResult.java
+++ b/features/sharding/core/src/main/java/org/apache/shardingsphere/sharding/merge/dal/show/ShowTableStatusMergedResult.java
@@ -17,7 +17,6 @@
 
 package org.apache.shardingsphere.sharding.merge.dal.show;
 
-import com.google.common.base.MoreObjects;
 import org.apache.shardingsphere.infra.binder.context.statement.SQLStatementContext;
 import org.apache.shardingsphere.infra.executor.sql.execute.result.query.QueryResult;
 import org.apache.shardingsphere.infra.merge.result.impl.memory.MemoryMergedResult;
@@ -89,7 +88,7 @@ public final class ShowTableStatusMergedResult extends MemoryMergedResult<Shardi
         return BigInteger.ZERO.equals(numberBigInteger) ? BigInteger.ZERO : getNonNullBigInteger(sum).divide(numberBigInteger);
     }
 
-    private static BigInteger getNonNullBigInteger(Object value) {
+    private static BigInteger getNonNullBigInteger(final Object value) {
         return Optional.ofNullable(value)
                 .filter(BigInteger.class::isInstance)
                 .map(BigInteger.class::cast)

